src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2040,19 +2040,21 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
       if ( mPrimaryKeyAttrs.size() == 1 &&
            defaultValueClause( mPrimaryKeyAttrs[0] ).startsWith( "nextval(" ) )
       {
-        bool foundNonNullPK = false;
+        bool foundNonEmptyPK = false;
         int idx = mPrimaryKeyAttrs[0];
         for ( int i = 0; i < flist.size(); i++ )
         {
           QgsAttributes attrs2 = flist[i].attributes();
           QVariant v2 = attrs2.value( idx, QVariant( QVariant::Int ) );
-          if ( !v2.isNull() )
+          // a PK field with a sequence val is auto populate by QGIS with this default
+          // we are only interested in non default values
+          if ( !v2.isNull() && !v2.toString().startsWith( "nextval(" ) )
```

Comment:
What happens if someone wants to save the string `nextval() is the function which is used by postgres to create a incrementing serial` in a field?

Reply:
You're right, I have changed the condition to exactly match the SQL default of the column so skipping the column in the INSERT will be absolutely equivalent